### ISSUE TITLE
Add option to disable public access

### ISSUE THREAD

**Issue by flunderpero** [GoogleCloudPlatform/k8s-config-connector#498]
See https://cloud.google.com/storage/docs/access-control/making-data-public

**Comment by maqiuyujoyce** [comment]
Hi @flunderpero , thank you for your question. Could you clarify a bit more about your scenario?
1. Do you currently have a public GCS bucket and its permissions are not managed by Config Connector, and you want to disable the public access via Config Connector?
2. If you have a public GCS bucket, is the permission configured via ACLs or IAM (i.e. [uniform bucket-level access](https://cloud.google.com/storage/docs/uniform-bucket-level-access))?

**Comment by flunderpero** [comment]
Hi @maqiuyujoyce, I just want to be able to set the flag that prevents a bucket from being made public by accident (see link above). Currently, I just set it by hand for the buckets that contain the most sensitive information and the config-connector reconciliation process leaves that flag untouched.

Right now, there is no way to specify that flag via config-connector.

**Comment by jcanseco** [comment]
> I just want to be able to set the flag that prevents a bucket from being made public by accident (see link above)

Hi @flunderpero, could you perhaps be talking about a different page? The linked page seems to only be talking about how to make objects in a bucket publicly accessible, but not about preventing buckets from being made public by accident (though we might just be missing something).

Perhaps you might have meant [this page](https://cloud.google.com/storage/docs/using-public-access-prevention) instead?

If so, then by a "flag that prevents a bucket from being made public by accident", are you perhaps referring to the `To enforce public access prevention at the bucket level` part of that page?

If that is the case, then I believe what you're looking for is support for the `iamConfiguration.publicAccessPrevention` field [here](https://cloud.google.com/storage/docs/json_api/v1/buckets).

Given that this resource is one of our old Terraform-based resources, I'll go ahead and just bump [this Terraform request](https://github.com/hashicorp/terraform-provider-google/issues/9519) which is requesting support for that very same field.

**Comment by flunderpero** [comment]
@jcanseco You are totally right about everything! Sorry, that I was not able to be precise enough.

**Comment by jcanseco** [comment]
No problem at all @flunderpero! Happy to help :)

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
